connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/DeadLetterQueueReporter.java

```diff
@@ -199,6 +199,10 @@ void populateContextHeaders(ProducerRecord<byte[], byte[]> producerRecord, Proce
     }
 
     private byte[] toBytes(String value) {
-        return value.getBytes(StandardCharsets.UTF_8);
+        if (value != null) {
+            return value.getBytes(StandardCharsets.UTF_8);
+        } else {
+            return null;
```

Comment:
should this return an empty array?

Reply:
Thanks for looking into this. Since the empty String gets serialized to an empty array, this would make null and the empty String indistinguishable and presumably deserialized into the empty String in the consumers. I think it's worth keeping null as a true null, since the RecordHeader class seems to accept null values and have careful handling of them.